Config/Config.cs

```diff
@@ -95,21 +95,22 @@ public void Merge(Config other)
 
         public void RebuildPatterns()
         {
-            HashSet<string> trustedAddressList = GetHashSet(TrustedDomains.Where(_ => _.Contains("@")));
-            HashSet<string> trustedDomainList = GetHashSet(TrustedDomains.Where(_ => !_.Contains("@")));
-            HashSet<string> unsafeAddressList = GetHashSet(UnsafeDomains.Where(_ => _.Contains("@")));
-            HashSet<string> unsafeDomainList = GetHashSet(UnsafeDomains.Where(_ => !_.Contains("@")));
-
-            TrustedDomainsPattern = $"^({string.Join("|", trustedDomainList.Select(ConvertWildCardToRegex))})$";
-            TrustedAddressesPattern = $"^({string.Join("|", trustedAddressList.Select(ConvertWildCardToRegex))})$";
-            UnsafeDomainsPattern = $"^({string.Join("|", unsafeDomainList.Select(ConvertWildCardToRegex))})$";
-            UnsafeAddressesPattern = $"^({string.Join("|", unsafeAddressList.Select(ConvertWildCardToRegex))})$";
-            UnsafeFilesPattern = $"({string.Join("|", UnsafeFiles.Select(ConvertWildCardToRegex))})";
+            TrustedAddressesPattern = $"^{ConvertListToMatcherRegex(TrustedDomains.Where(_ => _.Contains("@")))}$";
+            TrustedDomainsPattern = $"^{ConvertListToMatcherRegex(TrustedDomains.Where(_ => !_.Contains("@")))}$";
+            UnsafeAddressesPattern = $"^{ConvertListToMatcherRegex(UnsafeDomains.Where(_ => _.Contains("@")))}$";
+            UnsafeDomainsPattern = $"^{ConvertListToMatcherRegex(UnsafeDomains.Where(_ => !_.Contains("@")))}$";
+            UnsafeFilesPattern = ConvertListToMatcherRegex(UnsafeFiles);
+
+            QueueLogger.Log($"TrustedAddressesPattern = {TrustedAddressesPattern}");
+            QueueLogger.Log($"TrustedDomainsPattern = {TrustedDomainsPattern}");
+            QueueLogger.Log($"UnsafeAddressesPattern = {UnsafeAddressesPattern}");
+            QueueLogger.Log($"UnsafeDomainsPattern = {UnsafeDomainsPattern}");
+            QueueLogger.Log($"UnsafeFilesPattern = {UnsafeFilesPattern}");
         }
 
-        private HashSet<string> GetHashSet(IEnumerable<string> list)
+        private static string ConvertListToMatcherRegex(IEnumerable<string> list)
```

Comment:
```suggestion
        private static string ConvertToMatcherRegex(IEnumerable<string> list)
```

細かい話で恐縮ですが、引数で型を明示しているので、`List`は不要かと思いました。

Reply:
メモ: 引数は`List`型ではないが、一般的に一覧という意味での`list`であれば意味が通るので、`list`のままで問題なさそう